From 378968c66f90a1bac367cf08fa2ca9f421b56a77 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 4 Feb 2018 11:35:02 +0200
Subject: [PATCH 2221/2241] net: mvpp2x: change mv_pp2x_aggr_tx_queue struct
 order

Change of mv_pp2x_aggr_tx_queue struct order improve performance.

Most used variables in data path moved to  mv_pp2x_aggr_tx_queue
struct start. This allow to keep this varibles in cache and improve
performance.

Change-Id: I339c32e1049a301359aa68682117a77c5bffbddf
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/49979
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h | 25 +++++++++++--------------
 1 file changed, 11 insertions(+), 14 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
index 56e337d..2ec0a48 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
@@ -379,9 +379,6 @@ struct mv_pp2x_tx_queue {
 };
 
 struct mv_pp2x_aggr_tx_queue {
-	/* Physical number of this Tx queue */
-	u8 id;
-
 	/* Number of Tx DMA descriptors in the descriptor ring */
 	int size;
 
@@ -391,28 +388,28 @@ struct mv_pp2x_aggr_tx_queue {
 	/* Number of currently used Tx DMA descriptor in the descriptor ring used by HW */
 	int hw_count;
 
-	/* Virtual pointer to address of the Aggr_Tx DMA descriptors
-	* memory_allocation
-	*/
-	void *desc_mem;
+	/* Index of the next Tx DMA descriptor to process */
+	int next_desc_to_proc;
 
 	/* Virtual pointer to address of the Aggr_Tx DMA descriptors array */
 	struct mv_pp2x_tx_desc *first_desc;
 
-	/* DMA address of the Tx DMA descriptors array */
-	dma_addr_t descs_phys;
-
 	/* Index of the last Tx DMA descriptor */
 	int last_desc;
 
 	/* AGGR TX queue lock */
 	spinlock_t spinlock;
 
-	/* Index of the next Tx DMA descriptor to process */
-	int next_desc_to_proc;
+	/* DMA address of the Tx DMA descriptors array */
+	dma_addr_t descs_phys;
 
-	/* XPS mask */
-	cpumask_t affinity_mask;
+	/* Virtual pointer to address of the Aggr_Tx DMA descriptors
+	* memory_allocation
+	*/
+	void *desc_mem;
+
+	/* Physical number of this Tx queue */
+	u8 id;
 } __aligned(MVPP2_CACHE_LINE_SIZE);
 
 struct mv_pp2x_rx_queue {
-- 
2.7.4

